Makes/MakesViewController.m

```diff
@@ -0,0 +1,111 @@
+//
+//  ViewController.m
+//  Makes
+//
+//  Created by Mohamed Ragab on 1/8/17.
+//  Copyright © 2017 Mohamed Ragab. All rights reserved.
+//
+
+#import "MakesViewController.h"
+#import "Make.h"
+#import <AFNetworking/AFNetworking.h>
+#import <Masonry/Masonry.h>
+#import "MakesCollectionViewCell.h"
+#import "MakeDetailViewController.h"
+#import "MakesManager.h"
+#import <SDWebImage/UIImageView+WebCache.h>
+
+
+
+
+@interface MakesViewController ()
+@property (nonatomic, strong) NSArray *makes;
+@property (nonatomic, strong) UICollectionView *makesCollectionView;
+@end
+
+@implementation MakesViewController
+
+- (void)loadMakes {
+    
+    [[MakesManager manager] makesWithsuccess:^(NSArray *makes) {
+        self.makes = makes;
+        [self.makesCollectionView reloadData];
+    } failure:^(NSError *error) {
+        
+    }];
+
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view, typically from a nib.
+    self.makes = [[NSMutableArray alloc]init];
+    self.title = @"Makes";
+    
+    self.view.backgroundColor = [UIColor whiteColor];
+    
+    
+    UICollectionViewFlowLayout *layout=[[UICollectionViewFlowLayout alloc] init];
+    self.makesCollectionView =[[UICollectionView alloc] initWithFrame:self.view.frame collectionViewLayout:layout];
+    [self.makesCollectionView setDataSource:self];
+    [self.makesCollectionView setDelegate:self];
+    UIImageView *backgroundImage = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"tire_pattern"]];
+    self.view.backgroundColor = [UIColor blackColor];
+    self.makesCollectionView.backgroundView = backgroundImage;
+    
+
+    
+    [self.makesCollectionView setBackgroundColor:[UIColor whiteColor]];
+    [self.makesCollectionView registerClass:[MakesCollectionViewCell class] forCellWithReuseIdentifier:@"cellIdentifier"];
+    [self.view addSubview:self.makesCollectionView];
+    
+    
+    UIEdgeInsets padding = UIEdgeInsetsMake(0, 0, 0, 0);
+    
+    [self.makesCollectionView mas_makeConstraints:^(MASConstraintMaker *make) {
+        make.edges.equalTo(self.view).with.insets(padding);
+    }];
+    
+    [self loadMakes];
+
+    
+}
+
+- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section
+{
+    return self.makes.count;
+}
+
+
+- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
+    
+    MakesCollectionViewCell *cell=[collectionView dequeueReusableCellWithReuseIdentifier:@"cellIdentifier" forIndexPath:indexPath];
+    
+    cell.makeImageView.image = nil;
```

Comment:
why you set the image to nil here ?
if it has the cashed images you should set in to nil but not here you should implement prepare for reuse method in the cell calss